File: src/test/java/console/MainTest.java
package console;

import org.junit.Test;

import static console.Main.fileIsDetected;
import static console.Main.main;
import static org.junit.Assert.assertEquals;
import static org.junit.Assert.assertFalse;
import static org.junit.Assert.assertTrue;

public class MainTest {

    @Test
    public void exceptionTest() {

        try {
            main("-r", "-d", "src/test/testDir", "testFile");
            assertTrue(fileIsDetected);
        }
        catch (IllegalArgumentException e){
            System.out.println(e.getMessage());
        }
        try {
            main("-r", "-d", "src/test/testDir");
            assertFalse(fileIsDetected);
        } catch (IllegalArgumentException e) {
            assertEquals("please, enter a file name!", e.getMessage());
        }
        try {
            main("-r", "-d", "some dir", "testFile");
            assertFalse(fileIsDetected);
        } catch (IllegalArgumentException e) {
            assertEquals("this directory doesn't exist", e.getMessage());
        }
        try {
            main("-r", "-d", "src/test/testDir", "testFile", "testFile");
            assertFalse(fileIsDetected);
        } catch (IllegalArgumentException e) {
            assertEquals("please, enter a ONE file name!", e.getMessage());
        }
        try {
            main("-d", "src/test/testDir/testDir2", "testFile");
            assertFalse(fileIsDetected);
        } catch (IllegalArgumentException e) {
            assertEquals("no results for: testFile", e.getMessage());
        }
    }

}

File: src/test/kotlin/console/MainTest.kt
package console

import com.beust.jcommander.ParameterException
import logic.FindTest
import org.junit.jupiter.api.Assertions.assertEquals
import org.junit.jupiter.api.Assertions.assertThrows
import org.junit.jupiter.api.Test
import java.io.File
import java.nio.file.Paths

class MainTest {

    @Test
    fun mainTest() {
        FindTest().generateTestFiles()

        assertThrows(IllegalArgumentException::class.java, { start(arrayOf("code.kt", "code.kt")) }, "You need to specify one file name!")
        assertThrows(ParameterException::class.java, { start(arrayOf("-d", "code.kt")) }, "Main parameters are required (\"File name for finding\")")
        assertThrows(IllegalArgumentException::class.java, { start(arrayOf("-r", "-d", "code.kt", "code.kt")) }, "Given directory doesn't exist!")
        assertThrows(Exception::class.java, { start(arrayOf("code.kt")) }, "No file with that file name!")

        val s = File.separator
        val testdir = Paths.get("").toRealPath().toString() + "${s}src${s}test${s}testdir"
        assertEquals(testdir + "${s}programming${s}kotlin${s}code.kt" + "\n" +
                testdir + "${s}programming${s}kotlin${s}deeper${s}code.kt",
                start(arrayOf("-r", "-d", testdir + "${s}programming", "code.kt")))

        FindTest().deleteTestFiles()
    }
}